fdbserver/MoveKeys.actor.cpp

```diff
@@ -1282,13 +1284,19 @@ ACTOR static Future<Void> startMoveShards(Database occ,
 				// Keep track of shards for all src servers so that we can preserve their values in serverKeys
 				state std::unordered_map<UID, std::vector<Shard>> physicalShardMap;
 
-				tr.trState->taskID = TaskPriority::MoveKeys;
-				tr.setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
-				tr.setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+				tr->getTransaction().trState->taskID = TaskPriority::MoveKeys;
+				tr->setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
+				tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
 
-				wait(checkMoveKeysLock(&tr, lock, ddEnabledState));
+				wait(checkMoveKeysLock(&(tr->getTransaction()), lock, ddEnabledState));
 
-				Optional<Value> val = wait(tr.get(dataMoveKeyFor(dataMoveId)));
+				if (!loadedTssMapping) {
```

Comment:
nit: move this `readTSSMappingRYW()` block to line 1335, after we evaluate the `DataMoveMetaData` since we could exit early and end up not using the loaded `TssMapping`.

Reply:
Good one.